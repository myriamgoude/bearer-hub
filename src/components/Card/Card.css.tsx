import { css } from 'emotion'
import { getEmSize } from '../../styles/mixins'

export default {
  root: css`
    background: white;
    box-shadow: 0 2px 4px 0 rgba(3, 13, 54, 0.26);
    border-radius: 6px;
  `,
  paddingRegular: css`
    padding: ${getEmSize(24)};
  `,
  paddingLarge: css`
    padding: ${getEmSize(24)} ${getEmSize(48)};
  `,
  heightSmall: css`
    height: 100%;
    max-height: 240px;
  `
}
