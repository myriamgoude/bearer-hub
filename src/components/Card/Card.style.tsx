import { css } from '@emotion/core'
import { getEmSize } from '../../styles/mixins'
import { breakpoints } from '../../styles/variables'

export default {
  root: css`
    background: white;
    box-shadow: 0 2px 4px 0 rgba(3, 13, 54, 0.26);
    border-radius: 6px;

    h5 {
      @media (max-width: ${breakpoints.lg}) {
        font-size: 18px !important;
      }
    }
  `,
  paddingRegular: css`
    padding: ${getEmSize(24)};
  `,
  paddingLarge: css`
    padding: ${getEmSize(24)} ${getEmSize(48)};
  `,
  heightSmall: css`
    height: 100%;
  `
}
