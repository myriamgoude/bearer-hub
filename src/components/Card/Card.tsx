import * as React from 'react'
import { css, cx } from 'emotion'

import Link from '../Link'
import Text from '../Text'

import styles from './Card.css'

interface ICardProps {
  children?: object
  className?: string
  onClick?: any
  title?: string | any
  text?: any
  icon?: string
  padding?: string
  style?: string | object
  link?: string
  small?: boolean
}

const Card = (props: ICardProps) => {
  return props.link ? (
    <Link
      to={props.link}
      className={cx(
        css`
          &:hover {
            text-decoration: none;
          }
        `
      )}
    >
      <div
        onClick={props.onClick}
        className={cx(
          props.className && props.className,
          styles.root,
          props.padding === 'large' ? styles.paddingLarge : styles.paddingRegular,
          props.small && styles.heightSmall
        )}
      >
        {props.icon && <img src={`${props.icon}`} alt={`Icon for ${props.title}`} />}
        {props.title && <h3>{props.title}</h3>}
        {props.text && <Text text={props.text} />}
        {props.children && props.children}
      </div>
    </Link>
  ) : (
    <div
      onClick={props.onClick}
      className={cx(
        props.className && props.className,
        styles.root,
        props.padding === 'large' ? styles.paddingLarge : styles.paddingRegular,
        props.small && styles.heightSmall
      )}
    >
      {props.icon && <img src={`${props.icon}`} alt={`Icon for ${props.title}`} />}
      {props.title && <h3>{props.title}</h3>}
      {props.text && <Text text={props.text} />}
      {props.children && props.children}
    </div>
  )
}

export default Card
