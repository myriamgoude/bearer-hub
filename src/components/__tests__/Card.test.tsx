import * as React from 'react'
import renderer from 'react-test-renderer'
import Card from '../Card/Card'

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card title="foo" text="foo" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders with object values', () => {
    const tree = renderer
      .create(
        <Card
          title={
            <>
              <img src="#" className="foo" alt="icon" />
              <span>foo bar</span>
            </>
          }
          text={
            <>
              <span>foo bar</span>
            </>
          }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
