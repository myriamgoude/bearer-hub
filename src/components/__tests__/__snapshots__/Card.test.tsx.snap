// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="root paddingRegular"
>
  <h3>
    foo
  </h3>
  <p
    className="root"
  >
    foo
  </p>
</div>
`;

exports[`Header renders with object values 1`] = `
<div
  className="root paddingRegular"
>
  <h3>
    <img
      alt="icon"
      className="foo"
      src="#"
    />
    <span>
      foo bar
    </span>
  </h3>
  <p
    className="root"
  >
    <span>
      foo bar
    </span>
  </p>
</div>
`;
